sdk/src/types/block/macro.rs

```diff
@@ -36,17 +36,17 @@ macro_rules! impl_id {
             $len_vis const LENGTH: usize = $length;
 
             #[doc = core::concat!("Creates a new [`", core::stringify!($hash_name), "`].")]
-            $hash_vis fn new(bytes: [u8; Self::LENGTH]) -> Self {
-                Self::from(bytes)
+            $hash_vis const fn new(bytes: [u8; $hash_name::LENGTH]) -> Self {
```

Comment:
I thought Clippy preferred Self over the actual type name?

Reply:
whoops